modules/top-level/test.nix

```diff
@@ -8,8 +8,72 @@
 let
   cfg = config.test;
 
-  inherit (config) warnings;
-  assertions = builtins.concatMap (x: lib.optional (!x.assertion) x.message) config.assertions;
+  # Expectation submodule used for checking warnings and/or assertions
+  expectation = lib.types.submodule (
+    { config, ... }:
+    {
+      options = {
+        predicate = lib.mkOption {
+          type = with lib.types; functionTo bool;
+          description = ''
+            A predicate of that determines whether this expectation is met.
+
+            Type
+            ```
+            [String] -> Boolean
+            ```
+
+            Parameters
+            1. values - all warnings or matched assertions
+          '';
+        };
+        expect = lib.mkOption {
+          type = with lib.types; nullOr (enum (builtins.attrNames cfg.predefinedExpectationPredicates));
+          description = ''
+            If non-null, will be used together with `value` to define `predicate`.
+          '';
+          default = "any";
```

Comment:
I'm not too keen on having a default value for this field, it requires remembering (or searching) for the default

Reply:
This is actually a bug; the default should be `null`

Reply:
Then the str coercion (if kept) will need to be updated right?